packages/performance-tests/test/TransformerRegression.test.ts

```diff
@@ -102,17 +107,60 @@ async function runRegressionTests() {
   describe("Transformer Regression Tests", function () {
     testIModels.forEach(async (iModel) => {
       describe(`Transforms of ${iModel.name}`, async () => {
+        let sourceDb: BriefcaseDb;
+        let reportInfo: ReporterInfo
+
+        before(async () => {
+          Logger.logInfo(loggerCategory, `processing iModel '${iModel.name}' of size '${iModel.tShirtSize.toUpperCase()}'`);
+          sourceDb = await iModel.load();
+          const fedGuidSaturation = sourceDb.withStatement(
+            `
+            SELECT
+            CAST(SUM(hasGuid) as DOUBLE)/SUM(total) ratio 
+            FROM (
+              SELECT IIF(FederationGuid IS NOT NULL, 1, 0) hasGuid,
+              1 as total FROM bis.Element
+            )`,
+            (stmt) => {
+              assert(stmt.step() === DbResult.BE_SQLITE_ROW);;
+              return stmt.getValue(0).getDouble();
+            }
+          );
+          Logger.logInfo(loggerCategory, `Federation Guid Saturation '${fedGuidSaturation}'`);
+          const toGb = (bytes: number) => `${(bytes / 1024 **3).toFixed(2)}Gb`;
+          const sizeInGb = toGb(fs.statSync(sourceDb.pathName).size);
+          Logger.logInfo(loggerCategory, `loaded (${sizeInGb})'`);
+          reportInfo = {
+            "Id": iModel.iModelId,
+            "T-shirt size": iModel.tShirtSize,
+            "Gb size": sizeInGb,
+            "Branch Name": branchName,
+            "Federation Guid Saturation 0-1": fedGuidSaturation,
+          };
+        });
+
         testCasesMap.forEach(async (testCase, key) => {
+          before(async () => {
+            if(!sourceDb.isOpen){
```

Comment:
still not sure why we need this code? We know that we close it, we should always have to reopen it. I'd prefer we replace the `testIModel.load()` function with like `testIModel.getFilePath()` and just always open+close a fresh one using that, never loading it like we used to.

Reply:
I had this if because on the first test the db would always be open. So then I didn't want to call open again if it already was. But removing load would fix this problem, so I'll do that

Reply:
Having trouble with this. I don't fully understand the getFilePath() function's purpose. I've tried something else instead that I think works, and when debugging everything looks in order and I can collect the FedGuid saturation and imodel size. However I think its trying to run the tests before first running the code inside the `before()` stmt, so the sourcedb is never initialized. I could be wrong, but when debugging, a breakpoint in line 154 gets reached before 122.

Reply:
and the error is `TypeError: Cannot read properties of undefined (reading 'getFilePath')`, caused by ln 157 of this file